src/unknown_0255E0.c

```diff
@@ -1051,8 +1174,153 @@ void func_8003093C(s32 arg0) {
     gDPSetFogColor(D_8011B0A0++, D_8011D388[arg0].unk0 >> 0x10, D_8011D388[arg0].unk4 >> 0x10, D_8011D388[arg0].unk8 >> 0x10, 0xFF);
     gSPFogPosition(D_8011B0A0++, D_8011D388[arg0].unkC >> 0x10, D_8011D388[arg0].unk10 >> 0x10);
 }
+
+#ifdef NON_EQUIVALENT
+//unk8011D388 D_8011D388[4];
+
+typedef struct Object_64_FogChanger {
+    s16 unk0;
+} Object_64_FogChanger;
+
+typedef struct Object_3C_FogChanger {
+    u8 pad0[0x2];
+    s16 unk2;
+    s16 unk4;
+    s16 unk6;
+    s8 unk8;
+    u8 unk9;
+    u8 unkA;
+    u8 unkB;
+    s16 unkC;
+    s16 unkE;
+    s16 unk10;
+} Object_3C_FogChanger;
+
+void obj_loop_fogchanger(Object* obj) {
+    s32 temp_v0;
+    s32 index;
+    s32 sp74;
+    s32 i;
+    s32 phi_a0, phi_v1;
+    f32 x, z;
+    s32 temp_a3, temp_t0, temp_t1, temp_a1;
+    s32 temp;
+    Object **sp40;
+    Object_3C_FogChanger *sp44;
+    Object *tempObj;
+    unk8011D388 *temp_v0_3;
+    ObjectSegment *phi_s3;
+    
+    sp40 = NULL;
+    sp44 = obj->segment.unk3C_a.unk3C;
+    phi_s3 = NULL;
+    
+    // func_80066510() returns the bool D_80120D14. 1 if camera is controlled by cutscene, 0 if controlled by racer.
+    if (func_80066510()) {
+        phi_s3 = func_80069D7C();
+        sp74 = get_viewport_count() + 1;
+    } else {
+        sp40 = get_object_struct_array(&sp74);
+    }
+    
+    for(i = 0; i < sp74; i++) {
+        index = -1;
+        if (sp40 != NULL) {
+            tempObj = sp40[i]; // Might be a player object?
+            temp_v0 = ((Object_64_FogChanger*)tempObj->unk64)->unk0;
+            if ((temp_v0 >= 0) && (temp_v0 < 4) && (obj != D_8011D388[i].unk34)) {
+                index = temp_v0;
+                x = tempObj->segment.trans.x_position;
+                z = tempObj->segment.trans.z_position;
+            }
+        } else if ((i < 4) && (obj != D_8011D388[i].unk34)) {
```

Comment:
MAXCONTROLLERS? This one I'm not certain about, but it seems i is related to sp74, which is from get_viewport_count. This means it might be related to the number of active players. Not sure though. This is just a quick review.